### PR TITLE
Elation find physician

### LINKED ISSUE #107: Elation: Find Physician
(note to everyone... if you write a description and convert to an issue before you save your description, you **lose** your description)

## Problem
It's difficult to find the `primaryPhysicianId` and `caregiverPracticeId` fields, which are necessary for various actions

## Solution
Find Physicians action

## Acceptance criteria

Leverage [the elation api documentation](https://docs.elationhealth.com/reference/find-physicians) for the uri

- Given the ability to add the Find Physicians action to the step
- When I add the Find Physicians action
- All three query parameters are available as fields (strings are fine, even though the npi number is technically a number)

- Given i run the Find Physicians action
- When the number of responses != 1
- Then provide an error with a message such as "Find Physicians returned {x} results, but the number of results must equal exactly 1."

- Given i run the find physicians action
- When the number of responses == 1
- Then add the following datapoints in onComplete (assuming the result is saved as `result`):

```ts
physicianId: result.id,
physicianFirstName: result.first_name,
physicianLastName: result.last_name,
physicianCredentials: result.credentials,
physician.email: result.email,
physicianNPI: result.npi,
physicianUserId: result.user_id
caregiverPracticeId: result.practice
```

Please make sure the physicianId is congruent with primaryPhysicianId (`primary_physician_id`) as it shows up in fields in other places in elation. Do the same with the `caregiver_practice_id`.

- Given Find Physicians works successfully
- When I use the provided physicianId or caregiverPracticeId datapoints as inputs for fields in other elation actions
- Then there are no errors resulting from an incorrect type

(fyi the `physicianUserId` datapoint can be used as the "author" field for the chart note)

Thanks!

### PR DESCRIPTION
**Pull request by michal-grzelak**
- resolves #107

### CODE REVIEW

**Review by bejoinka — CHANGES_REQUESTED** [review]
looks good, almost there. looks like you missed the query strings. fields should be `first_name`, `last_name`, and `npi`. they should be passed as query params in the get request to elation

**Comment by michal-grzelak** [comment]
@bejoinka 
> looks good, almost there. looks like you missed the query strings. fields should be `first_name`, `last_name`, and `npi`. they should be passed as query params in the get request to elation

Thanks for spotting this. I totally missed those 🙈

File: extensions/elation/actions/findPhysician.ts
Comment (bejoinka) [inline extensions/elation/actions/findPhysician.ts:98]: Should this line be `if (physiciansList.count !== 1)`?

File: extensions/elation/client.ts
Comment (bejoinka) [inline extensions/elation/client.ts:10]: i think we can remove `Find` if we're going to use ElationCollection, right?

Comment (bejoinka) [reply extensions/elation/client.ts:10]: at the very least, there's some duplicate here that can be removed.

File: extensions/elation/types/generic.ts
Comment (bejoinka) [inline extensions/elation/types/generic.ts:11]: ditto my comment above. `ElationCollection` and `Find` accomplish the same thing. Let's use just one. I like the way you defined it.

**Review by bejoinka — CHANGES_REQUESTED** [review]
A couple small adjustments here:
- [] Handle 0 responses appropriately
- [] clean up `ElationCollection` / `Find` duplication

File: extensions/elation/actions/findPhysician.ts
Comment (michal-grzelak) [reply extensions/elation/actions/findPhysician.ts:98]: Yes, sure 👍🏻

**Review by bejoinka — APPROVED** [review]
Looks good. I'm going to approve without validation of the fields because undefined values are removed from params in axios. (e.g. `params: {foo: undefined, bar: 'baz'}` evaluates to `?bar=baz`)

**Comment by bejoinka** [comment]
oh, @michal-grzelak , totally missed it, but let's conform to what you did with #114 in this as well, and then we can merge 👍

**Comment by michal-grzelak** [comment]
@bejoinka 
> oh, @michal-grzelak , totally missed it, but let's conform to what you did with #114 in this as well, and then we can merge 👍

Hi, in https://github.com/awell-health/awell-extensions/pull/115 there is everything finished with those changes as well. You can see that `elation-consistency` branch has a base of `elation-find-physician` branch.

**Review by bejoinka — APPROVED** [review]
This is in good shape to merge once we get approval from the business side.

**Comment by bejoinka** [comment]
Okay, we're good to merge as soon as we rebase and use the new extensions core package.

@michal-grzelak -- do you want to pick this one back up once you're free?

**Comment by michal-grzelak** [comment]
@bejoinka Sure, I can can take it when I finish the current task

**Comment by michal-grzelak** [comment]
@bejoinka This PR is ready to be merged from my side. Old code removed and using @awell-health/extensions-core

**Pull request merged**
